### DISCUSSION TITLE
Block Blob instead of Page Blob

### DISCUSSION THREAD

**Discussion by michaelseto** [olahallengren/sql-server-maintenance-solution#120] [Questions & Answers]
We are leveraging the database backup solution and trying to convert our backups in Azure to the 'Archive' access tier. However, it appears we cannot do that as the backups were uploaded as a 'Page Blob' type, when they should be 'Block Blob'.

Where would this be fixed?

Thank you

**Comment by olahallengren** [comment]
I am not sure that I understand this. Is there an issue with the DatabaseBackup stored procedure, when backing up to Azure Blob Storage?

**Comment by michaelseto** [comment]
I see the backups uploaded as page blobs and I want them uploaded as block blobs. That’s as straight forward as I can explain it.

**Comment by olahallengren** [comment]
How are you executing DatabaseBackup? Could you post the command with all parameters.

**Comment by michaelseto** [comment]
Yes let me get that for you, just a moment please

**Comment by michaelseto** [comment]
```
EXECUTE DBA.dbo.DatabaseBackup @AvailabilityGroups = 'AG_Name',
@URL = 'https://mybackups.blob.core.windows.net/full/',
@Credential = 'blob_credentials',
@BackupType = 'FULL',
@Compress = 'Y',
@Encrypt = 'Y',
@EncryptionAlgorithm = 'AES_256',
@ServerCertificate = 'Backup_Certificate'
```

**Comment by olahallengren** [comment]
Thank you. Which version of SQL Server are you running on?

**Comment by michaelseto** [comment]
Microsoft SQL Server 2016 (SP2) (KB4052908) - 13.0.5026.0 (X64)   Mar 18 2018 09:11:49   Copyright (c) Microsoft Corporation  Enterprise Edition (64-bit) on Windows Server 2016 Datacenter 10.0 <X64> (Build 14393: ) (Hypervisor)

**Comment by olahallengren** [comment]
You need to create a Shared Access Signature (and create a credential for that), in order to back up to block blobs

You should not use the ```@Credential``` parameter in DatabaseBackup then.

https://blogs.msdn.microsoft.com/sqlcat/2017/03/10/backing-up-a-vldb-to-azure-blob-storage/

**Reply by CarlosGzz-SQL** [reply]
is supported to do stripping backups using SAS?

**Reply by CarlosGzz-SQL** [reply]
found your post, thank you! 
https://gist.github.com/scheffler/7edd40f430235aab651fadcc7d191a89

**Comment by michaelseto** [comment]
Thank you, I will update you on our progress.

**Comment by bulentgucuk** [comment]
Here is a link that will help you with this.  The powershell script is very helpful.  Read the comments and supply values for the parameters.  At the end the powershell script prints the create credential t-sql script to execute in your sql server.  After that, as Ola said you can run the backups without the @credential parameter. 

https://docs.microsoft.com/en-us/sql/relational-databases/lesson-1-create-stored-access-policy-and-shared-access-signature?view=sql-server-2017

**Comment by olahallengren** [comment]
Microsoft has also improved the documentation here:

"Backup to block blob vs. page blob"
https://docs.microsoft.com/en-us/sql/relational-databases/backup-restore/sql-server-backup-to-url?view=sql-server-2017
